dali/base/dadfs.cpp

```diff
@@ -9137,20 +9130,17 @@ class CInitGroups
         if (!eps.ordinality())
             return NULL;
         Owned<IGroup> grp;
-        unsigned slavesPerNode = 0;
-        if (grp_thor == groupType)
-            slavesPerNode = cluster.getPropInt("@slavesPerNode", 1);
-        if (expand && slavesPerNode)
+        if (grp_thor != groupType)
+            expand = false;
+        if (expand)
         {
+            unsigned slavesPerNode = cluster.getPropInt("@slavesPerNode", 1);
+            unsigned channelsPerSlave = cluster.getPropInt("@channelsPerSlave", 1);
             SocketEndpointArray msEps;
-            for (unsigned s=0; s<slavesPerNode; s++)
+            for (unsigned s=0; s<(slavesPerNode*channelsPerSlave); s++)
```

Comment:
minor:  calculating
unsigned channelsPerNode = slavesPerNode*channelsPerSlave;
before the loop might be slightly clearer.  The compiler is likely to evaluate it once anyway though so no need to change.